From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Kooran Paul, Princy" <princy.kooran.paul@intel.com>
Date: Mon, 14 Apr 2025 14:16:05 +0800
Subject: [PATCH] drm/i915: Fix logic for GUC Process

This is to fix the incorrect GUC logic.
It checks whether a field in the message is 0 or something else.

Signed-off-by: Kooran Paul, Princy <princy.kooran.paul@intel.com>
Signed-off-by: Zawawi, Muhammad Zul Husni <muhammad.zul.husni.zawawi@intel.com>
---
 drivers/gpu/drm/i915/gt/uc/intel_guc_submission.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/gt/uc/intel_guc_submission.c b/drivers/gpu/drm/i915/gt/uc/intel_guc_submission.c
index c07ab3ce941bdc98247b831d88261a5550409617..9a4171ba3209a0ca9153b7027555e0b7b80116ad 100644
--- a/drivers/gpu/drm/i915/gt/uc/intel_guc_submission.c
+++ b/drivers/gpu/drm/i915/gt/uc/intel_guc_submission.c
@@ -3116,7 +3116,7 @@ int intel_guc_process_set_engine_sched_done(struct intel_guc *guc, const u32 *ms
 	if (len != GUC2HOST_SET_ENGINE_SCHED_DONE_MSG_LEN)
 		return -EPROTO;
 
-	if (FIELD_GET(GUC2HOST_SET_ENGINE_SCHED_DONE_MSG_0_MBZ, msg[0] != 0))
+	if (FIELD_GET(GUC2HOST_SET_ENGINE_SCHED_DONE_MSG_0_MBZ, msg[0]) != 0)
 		return -EPROTO;
 
 	decr_outstanding_submission_g2h(guc);
